Checkpoint 4

eto-enot opened the pull request:
PR содержит работы по пунктам четвертого чекпоинта:
- Сервис FastAPI
- Приложение Streamlit
- [Отчет](/ai24-team-15/cbr-press-release-classifier/blob/checkpoint-4/report.md)

Реализованные бонусные пункты:
- Функционал сравнения экспериментов
- Использование plotly для построения интерактивных графиков
- Система сбора логов ELK
- Решение развенуто в Yandex Cloud

Приложение доступно по адресу http://cbr-classifier.ddns.net/

Система сбора логов: http://cbr-classifier.ddns.net:5601/app/discover  реквизиты: guest / guest123

Closes #7
Closes #9
Closes #10
Closes #11

eto-enot committed:
[wip] docker

eto-enot committed:
fix liter warnings

KuBaN658 committed:
add eda to frontend

dkzmn committed:
fastapi start

KuBaN658 committed:
add wordclouds

eto-enot committed:
add elk

eto-enot committed:
use volumes

KuBaN658 committed:
add tsne

KuBaN658 committed:
add multipage

KuBaN658 committed:
add upload data

KuBaN658 committed:
add logs on analytics

KuBaN658 committed:
add download data from s3

KuBaN658 committed:
add fast analytics for actual data

dkzmn committed:
fastapi service v1

dkzmn committed:
add requirements

dkzmn committed:
fix readme

KuBaN658 committed:
fix get stopwords from nltk

KuBaN658 committed:
add dockerfile

eto-enot committed:
rename projects

eto-enot committed:
update docker compose config

KuBaN658 committed:
Merge remote-tracking branch 'origin/fastapi' into streamlit

RuslanKHAI committed:
Update tsne.py

RuslanKHAI committed:
Update tsne.py

KuBaN658 committed:
move utils config plots to tools

KuBaN658 committed:
Merge remote-tracking branch 'origin/streamlit' into streamlit

RuslanKHAI committed:
Update wordcloud.py

RuslanKHAI committed:
Update dinamic.py

RuslanKHAI committed:
Update pie.py

RuslanKHAI committed:
Update length_text.py

KuBaN658 committed:
add fit and predict pages

KuBaN658 committed:
fix codestyle

KuBaN658 committed:
fix codestyle

KuBaN658 committed:
fix codestyle

KuBaN658 committed:
delete model.py

KuBaN658 committed:
fix codestyle

KuBaN658 committed:
fix codestyle

KuBaN658 committed:
add check_style.sh

eto-enot committed:
update readme

eto-enot committed:
update readme

KuBaN658 committed:
fix url s3

KuBaN658 committed:
get api url from env

KuBaN658 committed:
fix name log file

eto-enot committed:
add deploy config

eto-enot committed:
add requirements.txt

eto-enot committed:
update filebeat config

eto-enot committed:
update docs

KuBaN658 committed:
add rotate logs

dkzmn committed:
add calc_metrics, sync, predict for release

dkzmn committed:
new tests

KuBaN658 committed:
add rotate logs

dkzmn committed:
add stopwords

dkzmn committed:
fix style, add stopwords

eto-enot committed:
add dockerignore

KuBaN658 committed:
Merge remote-tracking branch 'origin/fastapi' into streamlit

dkzmn committed:
add nltk

KuBaN658 committed:
change handler predict

eto-enot committed:
update docs

KuBaN658 committed:
add comparison of models

KuBaN658 committed:
fix codestyle

KuBaN658 committed:
add remove_all, get_data and sync_data

eto-enot committed:
update requirements.txt

eto-enot committed:
download stopwords

eto-enot committed:
Merge branch 'streamlit' into checkpoint-4

eto-enot committed:
Merge branch 'fastapi' into checkpoint-4

eto-enot committed:
add screencasts

KuBaN658 committed:
add report

eto-enot committed:
update readme

eto-enot committed:
update report

KuBaN658 committed:
add report.pdf

Dmitry426 commented:
Минусы оформления 
 1. У питон модулей и пакетов отсутсвуют  __init__  , соостветвенно модуль не будет распознаваться при билде (
     в проекте используется орга , и я так понимаю каждый сервис в итоге будет иметь свой репозиторий )
 2. Навзвания модулей не сосответсвуют Pep8 ( https://peps.python.org/pep-0008/#package-and-module-names)
    fastapi(уже есть такая библиотека установка проекта в usr/lib вызовет конфиликт) frontend (навание не несет смысла)
 3.  . Env можно найти в разных меcтах по всему проекту ( лучше рассмотреть один глобальный конфиг)
 4. В самих (fastapi/parser) нет подмодулей все .py файлы находятся в пакетах ( не оптимальная структура приводит к плохой читаемости)
 5. Стаил гайд проекта должен быть общий , увидел frontend/check_style.sh как линтуются остальные модули не понятно . 
    Во всех проектах на одном языке нужна общая структура стилей 
 6. Разные  менеджеры зависимостей (лучше определиться на этапе mvp с одним
  менеджером чтобы потом не писать разные пайплайны CI/CD). Если используете poerty то лучше использовать его везде 
  и не генерировать множетсво requirements.txt потенциально с разным скопом (генерация
   в целом имеет болячки и задумана для 'плавного' перевода легаси продуктов с pip на poetry с сохранаинием старого пайплайна, 
   т.к проект делается с нуля то лучше использовать все возможности поэтри(  скоп , сборка и тд))
   хинт по cборке в докер (https://github.com/orgs/python-poetry/discussions/1879)
Докер: 
    1.  В фастапи есть докер игнор ,  в остальных модулях его нет. (должен быть в каждом модуле)
    2.  Так не сработает  в композ   env_file:
                                    - .env_elk
    3.  Наиваная реализация логера (не прописана ротация) , работу с эластик проще реализовать 
    с родной библиотекой логирования https://pypi.org/project/ecs-logging/
    4. Зачем в такой реализации логстеш ( если логи будут парвильного формата то можно сразу в еластик )
    5 . Круто что сделана общая докер сеть для проекта ! но filebit и остальной elk логически на разных нодах , 
    можно файлбит в основной композ закинуть.

eto-enot committed:
Исправлены замечания

- сделан один общий файл .env
- сделана общая сеть и для проекта, и для elk
- добавлен dockerignore

eto-enot committed:
add presentation

eto-enot committed:
update presentation

KuBaN658 committed:
rename frontend to cbr_ui

KuBaN658 committed:
Revert "rename frontend to cbr_ui"

This reverts commit 55bd432c3d9a563fea8cc36343cdf2f23bcde3ad.

KuBaN658 committed:
rename frontend to cbr_ui

eto-enot committed:
update presentation

KuBaN658 committed:
remove .env from cbr_ui

eto-enot committed:
update presentation

KuBaN658 committed:
add requirements.txt to eda, ml, scraping

KuBaN658 committed:
remove poetry from project

eto-enot committed:
update docker compose config

eto-enot committed:
update checkpoint.md

KuBaN658 committed:
add description streamlit and upgrade baseline

eto-enot committed:
update presentation

KuBaN658 committed:
add EDA to presentation

eto-enot committed:
add images

eto-enot committed:
update presentation

KuBaN658 committed:
add zharkovskiy dmitriy to distribution of work in the team

eto-enot committed:
add summary

dkzmn committed:
add fastapi description

dkzmn committed:
Merge branch 'preliminary' of github.com:ai24-team-15/cbr-press-release-classifier into preliminary

eto-enot committed:
update presentation

RuslanKHAI committed:
Update preliminary.md

KuBaN658 committed:
remove todo

eto-enot committed:
Merge branch 'preliminary' into checkpoint-4

dkzmn committed:
fix some checkpoint comments

dkzmn committed:
fix some checkpoint comments

eto-enot commented:
Исправление замечаний отслеживается в #13
